app/image_recognizer.py

```diff
@@ -0,0 +1,154 @@
+        )
+
+    def download_default_models(self) -> None:
+        """Downloads defaults keras models
+
+        Raises:
+            self.MissingModelsLinksFile: Models links file not found
+            self.InvalidModelsLinksFile: Models links file is invalid
+            self.InvalidModelLink: Resolution model link is invalid
+            self.InvalidModelLink: Color format model link is invalid
+        """
+        try:
+            with open(self.MODELS_LINKS_FILE_PATH, "r", encoding="utf-8") as file:
+                links = json.load(file)
+                resolution_model_link = links[self.MODELS_LINKS_RESOLUTION_KEY]
+                color_format_model_link = links[self.MODELS_LINKS_COLOR_FORMAT_KEY]
+        except FileNotFoundError:
+            logging.error("Models links file not found")
+            raise self.MissingModelsLinksFile("Models links file not found")
+        except ValueError:
+            logging.error("Models links file is invalid")
+            raise self.InvalidModelsLinksFile("Models links file is invalid")
+
+        if gdown.download(resolution_model_link, self.DEFAULT_RESOLUTION_MODEL_PATH) is None:
+            logging.error("Resolution model link is invalid")
+            raise self.InvalidModelLink("Resolution model link is invalid")
+        if gdown.download(color_format_model_link, self.DEFAULT_COLOR_FORMAT_MODEL_PATH) is None:
+            logging.error("Color format model link is invalid")
+            raise self.InvalidModelLink("Color format model link is invalid")
+
+    def recognize(self, raw_img_path: str) -> Result:
+        """Recognize raw image - find correct color format and resolution
+
+        Args:
+            raw_img_path (str): path to raw image
+
+        Returns:
+            Result: object of Result class with found color format, resolution and confidences for them
+        """
+        print(f"Searching for the top {self.RESOLUTION_RESULTS_N} best resolutions...")
+        resolutions = self.resolution_finder.find_resolution(raw_img_path, best_results=self.RESOLUTION_RESULTS_N)
+        best_result = self.Result()
+
+        for resolution in resolutions:
+            print(f"Searching for the best color format for {resolution.width}x{resolution.height} resolution...")
+            color_formats_confidences = self.color_format_finder.find_color_format(raw_img_path, resolution.width)
+            best_color_format = max(color_formats_confidences, key=color_formats_confidences.get)
+            if color_formats_confidences[best_color_format] > best_result.color_format_confidence:
+                best_result.color_format = best_color_format
+                best_result.img_width = resolution.width
+                best_result.img_height = resolution.height
+                best_result.color_format_confidence = color_formats_confidences[best_color_format]
+                best_result.resolution_confidence = resolution.confidence
```

Comment:
Maybe instead picking result with best confidence, pick the one that occurs most often (counting for example only those with >95 confidence)

Reply:
I think that picking the best one is better, if detected resolution is incorrect, then color format confidence will be very low anyway, and then the most confident one (with for example 0.2 confidence) may have better chances for being the best, than one that is randomly repeating with even lower confidence score.

Reply:
Ye but i think those incorrect resolutions, may still look somehow correct. For example lets assume that we got 3 resolutions and for 2 of them we would get 96 and 97 for rgb. Then for third one lets have 98 for abgr. RGB i thnik in this case would be more accurate.

Reply:
Incorrect resolution will maybe have 0.2 confidence at best, so this approach would require results to have very low confidence threshold, which could lead to random results.